pyccel/ast/core.py

```diff
@@ -3149,90 +3149,135 @@ def __init__(
 
     @property
     def name(self):
+        """ Name of the function """
         return self._name
 
     @property
     def arguments(self):
+        """ List of variables which are the function arguments """
         return self._arguments
 
     @property
     def results(self):
+        """ List of variables which are the function results """
         return self._results
 
     @property
     def body(self):
+        """ CodeBlock containing all the statements in the function """
         return self._body
 
     @property
     def local_vars(self):
+        """ List of variables defined in the function """
         return self._local_vars
 
     @property
     def global_vars(self):
+        """ List of global variables used in the function """
         return self._global_vars
 
     @property
     def cls_name(self):
+        """ String containing the name of the class to which the method belongs.
+        If the function is not a class procedure then this returns None """
         return self._cls_name
 
     @property
     def hide(self):
+        """ False if function is pyccelized, True otherwise """
         return self._hide
```

Comment:
@saidctb Is this description correct? If so then why do we need a `FunctionDef` object for this function?

Reply:
we use it when we want to inline a function

Reply:
Is there a way of doing that currently in pyccel?

Reply:
we had it, but it needs to be updated, it's used when the user add `@inline` decorator

Reply:
But we don't have an `@inline` decorator